front/src/views/leisure/leisure.view.js

```diff
@@ -15,32 +16,175 @@ export default {
+      google: null,
+      map: null,
+      apiKey: 'AIzaSyDuiiLqO-HTj8vP0POPrsBGJdWc7esHdOY',
+      radius: 1,
+      paginationLength: 10,
+      totalVisible: 9,
+      page: 1
     };
   },
-  mounted () {}
+  computed: {
+    mapConfig () {
+      return {
+        ...mapSettings,
+        zoom: 13
+      };
+    }
+  },
+  methods: {
+    async initializeMap () {
+      this.google = await GoogleMapsApiLoader({
+        libraries: [ 'places' ],
+        apiKey: this.apiKey
+      });
+
+      this.map = new this.google.maps.Map(this.$refs['leisure-map'], this.mapConfig);
+      this.placesService = new this.google.maps.places.PlacesService(this.map);
+
+      const container = document.getElementById('leisure-autocomplete');
+      this.autocomplete = new this.google.maps.places.Autocomplete(container);
+      this.autocomplete.addListener('place_changed', this.onPlaceChanged);
+
+      navigator.geolocation.getCurrentPosition(this.setMapCenter);
+    },
+    setMapCenter (position) {
+      if (position) {
+        const pos = {
+          lat: position.coords.latitude,
+          lng: position.coords.longitude
+        };
+        this.mapCenter = pos;
+        this.map.setCenter(this.mapCenter);
+      } else {
+        this.map.setCenter(this.mapCenter);
+      }
+      this.getNearbyPlaces(this.mapCenter);
+    },
+    onPlaceChanged () {
+      const place = this.autocomplete.getPlace(),
+            position = {
+              lat: place.geometry.location.lat(),
+              lng: place.geometry.location.lng()
+            };
+      if (place.geometry) {
+        this.map.panTo(place.geometry.location);
+        this.map.setZoom(15);
+      }
+      this.getNearbyPlaces(position);
+    },
+    async getNearbyPlaces (position) {
+      const search = {
+        bounds: this.map.getBounds(),
+        types: this.leisureTypes,
+        location: position,
+        radius: this.radius * 1000,
+        photo: null
+      };
+
+      this.placesService.nearbySearch(search, (results, status, pagination) =>
+        this.nearbySearchCallback(results, status, pagination));
+    },
+    nearbySearchCallback (places, status, pagination) {
+      this.clearMarkers();
+      this.places = [];
+      if (status === this.google.maps.places.PlacesServiceStatus.OK) {
+        places.forEach((place, i) => {
+          this.markers[i] = new this.google.maps.Marker({
+            position: place.geometry.location,
+            animation: this.google.maps.Animation.DROP
+          });
+
+          if (place.photos) {
+            place.photos = place.photos[0].getUrl();
+          }
+          if (place.opening_hours) {
+            if (place.opening_hours.open_now) {
+              place.opening_hours = this.$t('leisure.placeOpenState');
+            } else {
+              place.opening_hours = this.$t('leisure.placeCloseState');
+            }
+          }
+
+          this.getNextPage = pagination.hasNextPage && function () {
```

Comment:
what is it getNextPage? it seems you said that we don\t know total amount to use pagination